include/rocksdb/comparator.h

```diff
@@ -74,6 +74,11 @@ class Comparator {
                                               const Slice& /*t*/) const {
     return false;
   }
+
+  // return true if the comparator equity infers hash equity.
+  // The major use case is to determine if DataBlockHashIndex is compatible
+  // with the customized comparator.
+  virtual bool ComparatorEquityInfersHashEquity() const { return false; }
```

Comment:
Call it a better name. I was thinking of something like `CanDifferentBytesEqual()`. It's not very good either. Think about somethings.